tcl/is.tcl

```diff
@@ -534,10 +266,58 @@ namespace eval qc::is {
         return [expr {[string toupper $bool] in {Y N YES NO TRUE FALSE T F 0 1}}]
     }
 
-    proc decimal {number} {
+    proc decimal {args} {
         #| Checks if the given number is a decimal.
+        qc::args $args -precision ? -scale ? -- number
         if {[string is double -strict $number]} {
-            return 1
+            if { ! [info exists precision] && ! [info exists scale] } {
+                return 1
+            } elseif { ! [info exists precision] && [info exists scale] } {
+                set count [string length [lindex [split $number .] 1]]
+                if { $scale >= 0 && [qc::is integer $scale]} {
+                    return [expr {$scale >= $count}]
+                } else {
+                    return -code error "Scale must be a non-negative integer."
+                }
+            } elseif { [info exists precision] } {
+                if { $precision <= 0 || ! [qc::is integer $precision]} {
+                    return -code error "Precision must be a positive integer."
+                }
+                
+                set parts [split $number .]
+                set left_count [string length [lindex $parts 0]]
+                set right_count [string length [lindex $parts 1]]
+                if { $left_count == 0 } {
+                    # The leading zero wasn't given.
+                    set left_count 1
+                }
+
```

Comment:
Problem I can foresee here is that [PostgreSQL allows a massive 131072 digits before the decimal point](http://www.postgresql.org/docs/9.4/static/datatype-numeric.html#DATATYPE-NUMERIC-TABLE) so casting to integer would fail if it were out of integer range.